File: backend/childcheck/urls.py
from . import views

from django.urls import path, re_path, include
from django.views.generic.base import RedirectView
from rest_framework.routers import DefaultRouter

router = DefaultRouter()
router.register('children', views.ChildViewSet, basename='Child')
router.register('guardians', views.GuardianViewSet, basename='Guardian')
router.register('checks', views.CheckViewSet, basename='Check')
router.register('users', views.UserViewSet, basename='User')

app_view = RedirectView.as_view(url='/static/index.html', permanent=True)
favicon_view = RedirectView.as_view(url='/static/favicon.ico', permanent=True)

urlpatterns = [
    path('api/', include(router.urls)),
    path('api/login/', views.LoginView.as_view(), name='Login'),
    path('api/scan/', views.ScanView.as_view(), name='Scan'),
    re_path(r'^favicon\.ico$', favicon_view),
    re_path('', app_view),
    re_path(r'^(?:.*)/?$', app_view),
]

File: frontend/src/utility/urls.js
var DOMAIN;

if (process.env.NODE_ENV !== 'production') {
  DOMAIN = `${window.location.protocol}//${window.location.hostname}:8000`;
} else {
  DOMAIN = window.location.origin;
}

const formatId = (url, id) => {
  if (id !== null) {
    return `${url}${id}/`;
  } else {
    return url;
  }
};

const addQuery = (url, params) => {
  if (params) {
    for (const [key, value] of Object.entries(params)) {
      url.searchParams.append(key, value);
    }
  }
  return url;
};

const urls = {
  addQuery: addQuery,
  child: (id = null, params = null) => {
    var url = new URL(formatId(`${DOMAIN}/api/children/`, id));
    return addQuery(url, params);
  },
  guardian: (id = null, params = null) => {
    var url = new URL(formatId(`${DOMAIN}/api/guardians/`, id));
    return addQuery(url, params);
  },
  check: (id = null, params = null) => {
    var url = new URL(formatId(`${DOMAIN}/api/checks/`, id));
    return addQuery(url, params);
  },
  user: (id = null, params = null) => {
    var url = new URL(formatId(`${DOMAIN}/api/users/`, id));
    return addQuery(url, params);
  },
  scan: (params = null) => {
    var url = new URL(`${DOMAIN}/api/scan/`);
    return addQuery(url, params);
  },
  photo: (path) => {
    if (path) {
      if (path.includes("http")) {
        return path;
      } else {
        return `${DOMAIN}${path}`;
      }
    } else {
      return null;
    }
  },
  report: `${DOMAIN}/api/report/`,
  backup: `${DOMAIN}/api/backup/`,
  login: `${DOMAIN}/api/login/`,
};

export default urls;
